File: 프로그래머스/lv1/42840. 모의고사/모의고사.js
function solution(answers) {
    let one = [1,2,3,4,5];
    let two = [2,1,2,3,2,4,2,5];
    let thr = [3,3,1,1,2,2,4,4,5,5];
    
    let one2 = answers.filter((v,i)=> v===one[i%one.length]).length;
    let two2 = answers.filter((v,i)=> v===two[i%two.length]).length;
    let thr2 = answers.filter((v,i)=> v===thr[i%thr.length]).length;
    let max = Math.max(one2,two2,thr2);
    
    let answer = [];
    if(one2 === max) answer.push(1)
    if(two2 === max) answer.push(2)
    if(thr2 === max) answer.push(3)
    
    return answer
}

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
import java.util.stream.IntStream;

class Solution {
    public int[] solution(int[] answers) {
        List<Integer> answer = new ArrayList<>();

        int[] firstAnswers = {1,2,3,4,5};
        int[] secondAnswers = {2,1,2,3,2,4,2,5};
        int[] thirdAnswers = {3,3,1,1,2,2,4,4,5,5};


        int first = (int) IntStream.range(0, answers.length).filter(i -> firstAnswers[i % firstAnswers.length] == answers[i]).count();
        int second = (int) IntStream.range(0, answers.length).filter(i -> secondAnswers[i % secondAnswers.length] == answers[i]).count();
        int third = (int) IntStream.range(0, answers.length).filter(i -> thirdAnswers[i % thirdAnswers.length] == answers[i]).count();

        int max = Math.max(first, Math.max(second, third));

        if (max == first) {
            answer.add(1);
        }
        
        if (max == second) {
            answer.add(2);
        }
        
        if (max == third) {
            answer.add(3);
        }

        return answer.stream().mapToInt(Integer::valueOf).toArray();
    }
}